frontend/public/co-fetch.js

```diff
@@ -124,8 +124,8 @@ export const coFetchUtils = {
   parseJson,
 };
 
-export const coFetchJSON = (url, method = 'GET', options = {}) => {
-  const headers = { Accept: 'application/json' };
+export const coFetchCommon = (url, method = 'GET', options = {}, customHeaders = {}) => {
```

Comment:
I believe you can pass headers already through options, so we shouldn't need another parameter. We should change the logic not to override `Accept` below IMO, just provide JSON as the default if not specified.

Reply:
Ah right you are... this is a merge, that was my bad when I was originally suggesting this pattern.

```
  // Pass headers last to let callers to override Accept.
  const allOptions = _.defaultsDeep({ method }, options, { headers });
```
That merge should result in headers being applied together. I guess likely due to my other change in `coFetchText` caused this to pop up. @sahil143 please look to drop the last param and go with the additional option. It should likely work.